### PR TITLE
Configure ModiTect parent

### LINKED ISSUE #86: Configure publication to Maven Central using Github actions
Bintray is sunseting on May 1st 2021. We need an alternative

https://github.com/adobe/S3Mock/blob/main/.github/workflows/maven-release.yml

https://github.com/raphw/byte-buddy/blob/master/.github/workflows/main.yml#L93

https://bjansen.github.io/java/2021/02/03/deploying-to-maven-central-using-github-actions.html

### PR DESCRIPTION
**Pull request by aalmiray**
Uses the `moditect-org-parent` POM and JReleaser for configuring releases to Maven Central, GitHub, and Sdkman.

Fixes #86

### CODE REVIEW

File: examples/vertx-example/layrry-links-runner/src/test/java/org/moditect/layrry/example/links/test/LayrryLinksTest.java [hunk 1]
```diff
@@ -82,30 +82,30 @@ public void canAddAndRemovePlugin() throws Exception {
...
 
         FilesHelper.deleteFolder(preparedPluginDir.resolve(TOURNAMENT_PLUGIN_NAME), pluginDir1);
 
         await().atMost(30, TimeUnit.SECONDS).until(() -> {
             return given()
-            .when()
-                .get("/tournaments")
-            .then()
-                .extract().statusCode() == 404;
+                    .when()
```
Comment (gunnarmorling) [inline examples/vertx-example/layrry-links-runner/src/test/java/org/moditect/layrry/example/links/test/LayrryLinksTest.java:105]: Any chance to keep out those whitespace changes?

File: examples/vertx-example/layrry-links-tournament/src/main/java/module-info.java [hunk 2]
```diff
@@ -1,8 +1,5 @@
-import org.moditect.layrry.example.links.core.spi.RouterContributor;
```
Comment (gunnarmorling) [inline examples/vertx-example/layrry-links-tournament/src/main/java/module-info.java:1]: Any particular reason for changing this?

File: layrry-core/src/test/java/org/moditect/layrry/internal/resolver/FlatLocalRepositoryTest.java [hunk 3]
```diff
@@ -15,6 +15,12 @@
  */
 package org.moditect.layrry.internal.resolver;
 
+import java.io.File;
```
Comment (gunnarmorling) [inline layrry-core/src/test/java/org/moditect/layrry/internal/resolver/FlatLocalRepositoryTest.java:18]: Is that ordering change required?

**Review by gunnarmorling — COMMENTED** [review]
Thx! Some minor comments inline. And any chance for splitting this up into three commits:

* parent POM stuff
* year update in headers
* log4j version update?

File: examples/vertx-example/layrry-links-tournament/src/main/java/module-info.java (hunk 2 above)
Comment (aalmiray) [reply examples/vertx-example/layrry-links-tournament/src/main/java/module-info.java:1]: The license plugin would wipe out the imports and file header for no reason other than having those imports at the top of the file

File: layrry-core/src/test/java/org/moditect/layrry/internal/resolver/FlatLocalRepositoryTest.java (hunk 3 above)
Comment (aalmiray) [reply layrry-core/src/test/java/org/moditect/layrry/internal/resolver/FlatLocalRepositoryTest.java:18]: Hmm shouldn't that be checked by impsort?

**Comment by aalmiray** [comment]
Hmm I can give it a go to split the commit into smaller chunks (_rolls_sleeves_ oh wait it's past midnight 😪 ...)

**Comment by aalmiray** [comment]
@gunnarmorling split in 3 is done 😄

File: layrry-core/src/test/java/org/moditect/layrry/internal/resolver/FlatLocalRepositoryTest.java (hunk 3 above)
Comment (gunnarmorling) [reply layrry-core/src/test/java/org/moditect/layrry/internal/resolver/FlatLocalRepositoryTest.java:18]: Yeah, it should.

**Comment by gunnarmorling** [comment]
Applied. Thanks a lot! The split definitely helped to grok what's going on 🙏 .

File: .github/workflows/maven.yml [hunk 4]
```diff
@@ -28,54 +25,41 @@ on:
...
-          distribution: 'zulu'
 
-      - name: Cache Maven packages
-        uses: actions/cache@v2.1.7
-        with:
-          path: ~/.m2
-          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
-          restore-keys: ${{ runner.os }}-m2
+    - name: 'Build'
+      run: mvn install --file pom.xml
```
Comment (aalmiray) [inline .github/workflows/maven.yml:50]: should be `/.mvnw`

Comment (gunnarmorling) [reply .github/workflows/maven.yml:50]: Done.

**Pull request merged**